File: projectBlueWater/src/main/java/com/lasthopesoftware/storage/write/permissions/FileWritePossibleArbitrator.java
package com.lasthopesoftware.storage.write.permissions;

import com.lasthopesoftware.storage.RecursiveFileAssertionTester;

import java.io.File;

public final class FileWritePossibleArbitrator implements IFileWritePossibleArbitrator {

	@Override
	public boolean isFileWritePossible(File file) {
		return RecursiveFileAssertionTester.recursivelyTestAssertion(file, File::canWrite);
	}
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/storage/write/permissions/FileWritePossibleArbitrator.kt
package com.lasthopesoftware.storage.write.permissions

import com.lasthopesoftware.storage.walkUpUntilFileExists
import java.io.File

class FileWritePossibleArbitrator : IFileWritePossibleArbitrator {
	override fun isFileWritePossible(file: File): Boolean = file.walkUpUntilFileExists()?.canWrite() ?: false
}
